kneed/shape_detector.py

```diff
@@ -0,0 +1,21 @@
+import numpy as np
+
+
+def find_shape(x, y):
+    """
+    Detect the direction and curve type of the line.
+
+    :return: direction("increasing" or "decreasing") and curve type("concave" or "convex")
+    """
+
+    p = np.polyfit(x, y, deg=1)
+    x1, x2 = int(len(x) * 0.2), int(len(x) * 0.8)
+    q = np.mean(y[x1:x2]) - np.mean(x[x1:x2] * p[0] + p[1])
```

Comment:
What is q?

Reply:
sorry for this too late reply.
q means if we draw a straight line from the first point to the last point, whether the curve is above or below this straight line.